### PR TITLE
[AIRFLOW] added airflow preflight check dag doc

### PR DESCRIPTION
**Pull request by rahul-madaan**
DAG Code has been approved by @JDarDagran [here](https://github.com/OpenLineage/OpenLineage/pull/2519)

### Problem

People face difficulty when setting up OpenLineage in Airflow. There is a lack of an easy way to verify if OpenLineage has been setup correctly and is able to send events, which can lead to difficulties in debugging.

### Solution

Introducing a new DAG named "openlineage_preflight_check_dag" This DAG is designed to be executed by users who wish to verify their OpenLineage setup within Airflow. It serves as a diagnostic tool to ensure that OpenLineage has been integrated properly and is capable of sending events as expected. 

The beauty of this solution lies in its compatibility with all versions and distributions of Airflow, making it a versatile tool for users across different environments.

This DAG takes care of both openlineage-airflow library as well as apache-airflow-providers-openlineage package

I have created placeholders for areas which require community inputs and task to be executed in case lineage backend is not Marquez

#### One-line summary:

`openlineage_preflight_check_dag` DAG which can be executed if user need to check if OpenLineage has been set up correctly or not.

SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2024 contributors to the OpenLineage project

### CODE REVIEW

File: docs/integrations/airflow/preflight-check-dag.md [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+    verify_backend(LINEAGE_BACKEND, config)
+
+
+def verify_backend(backend_type: str, config: dict):
+    backend_type = backend_type.lower()
+    if backend_type == "marquez":
+        return _verify_marquez_http_backend(config)
+    elif backend_type == "atlan":
+        return _verify_atlan_http_backend(config)
+    elif backend_type == "marquez":
```
Comment (dolfinus) [inline docs/integrations/airflow/preflight-check-dag.md:276]: Probably just `else:`

Comment (rahul-madaan) [reply docs/integrations/airflow/preflight-check-dag.md:276]: Thanks! this got missed 😅

Comment (kacpermuda) [reply docs/integrations/airflow/preflight-check-dag.md]: Not sure i understand, where do we need `else` here? We should probably change "marquez" to "custom" tho, as it's duplicated now ;)

**Review by mobuchowski — APPROVED** [review]

**Comment by kacpermuda** [comment]
Thanks @rahul-madaan for the contribution !

**Pull request merged**